File: Tier3/src/main/java/via/sep3/group11/tier3/model/Refugee.java
package via.sep3.group11.tier3.model;

import lombok.Data;
import javax.persistence.*;
import java.time.LocalDate;

import static via.sep3.group11.tier3.model.Date.convertDateObjectToLocalDate;
import static via.sep3.group11.tier3.model.Date.convertLocalDateToDateObject;

/**
 * A class responsible for creating an Address table.
 * Contains an entity annotation that persists to the database
 * and Data annotation in order to inject all the necessary constructors and methods
 *
 * @author Group 11
 * @version 28/11/22
 */
@Data
@Entity
@Table(name = "refugee")
public class Refugee {

    @Id
    private String email;

    @Column(name = "fName")
    private String firstName;

    @Column(name = "mName")
    private String middleName;

    @Column(name = "lName")
    private String lastName;

    @Column(name = "password")
    private String password;

    @Column(name = "nationality")
    private String nationality;

    @Column(name = "gender")
    private char gender;

    @Column(name = "dob")
    private LocalDate dateOfBirth;

    @Column(name = "familySize")
    private int familySize;

    @Column(name = "description")
    private String description;

    public Refugee() {}

    public Refugee(String email, String firstName, String middleName, String lastName, String password, String nationality, char gender, Date dateOfBirth, int familySize, String description) {
        this.email = email;
        this.firstName = firstName;
        this.middleName = middleName;
        this.lastName = lastName;
        this.password = password;
        this.nationality = nationality;
        this.gender = gender;
        this.dateOfBirth = convertDateObjectToLocalDate(dateOfBirth);
        this.familySize = familySize;
        this.description = description;
    }

    public String getEmail() {
        return email;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getMiddleName() {
        return middleName;
    }

    public String getLastName() {
        return lastName;
    }

    public String getPassword() {
        return password;
    }

    public String getNationality() {
        return nationality;
    }

    public char getGender() {
        return gender;
    }

    public Date getDateOfBirth() {
        return convertLocalDateToDateObject(dateOfBirth);
    }

    public int getFamilySize() {
        return familySize;
    }

    public String getDescription() {
        return description;
    }
}

File: Program/Tier1/Shared/Domain/Refugee.cs
namespace Shared.Domain;

public class Refugee
{
    public string Email { get; set; }
    public string Password { get; set; }
    public char Gender { get; set; }
    public string Nationality { get; set; }
    public string FirstName { get; set; }
    public string MiddleName { get; set; }
    public string LastName { get; set; }
    
    public Date DateOfBirth { get; set; }

    public int FamilySize { get; set; }

    public string Description { get; set; }

    public Refugee(string email, string password, char gender, string nationality,
        string firstName, string middleName, string lastName, Date dateOfBirth, int familySize, string description)
    {
        Email = email;
        Password = password;
        Gender = gender;
        Nationality = nationality;
        FirstName = firstName;
        MiddleName = middleName;
        LastName = lastName;
        DateOfBirth = dateOfBirth;
        FamilySize = familySize;
        Description = description;
    }

    public Refugee()
    {
    }
}